src/main/java/com/netflix/imflibrary/writerTools/utils/SequenceDurationValidator.java

```diff
@@ -0,0 +1,67 @@
+package com.netflix.imflibrary.writerTools.utils;
+
+import com.netflix.imflibrary.st2067_2.IMFTrackFileResourceType;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class SequenceDurationValidator {
+
+    /**
+     * Validates and adjusts the duration of the resources in the sequence to ensure the total duration
+     * is an integer number of Composition Edit Units.
+     *
+     * @param resources the list of resources in the sequence
+     * @param compositionEditRateNumerator of the composition edit rate
+     * @param compositionEditRateDenominator of the composition edit rate
+     * @return a new list of resources with adjusted durations if needed
+     */
+    public static List<IMFTrackFileResourceType> validateAndAdjustResourceDurations(List<IMFTrackFileResourceType> resources, long compositionEditRateNumerator, long compositionEditRateDenominator) {
+        long totalResourceDuration = resources.stream()
+                .mapToLong(resource -> {
+                    if (resource != null) {
+                        return resource.getSourceDuration().longValue();
```

Comment:
I believe you should use `resource.getDuration()` instead of `getSourceDuration()`.
Although it is pretty uncommon, it is possible for a Resource to set a 'repeatCount'. The `getDuration` method will include the total for _all_ repeats.